python/src/space/core/ops/change_data.py

```diff
@@ -64,12 +70,9 @@ def read_change_data(
         f"Start snapshot ID not found: {end_snapshot_id}")
 
   for snapshot_id in all_snapshot_ids:
-    if snapshot_id <= start_snapshot_id:
+    if snapshot_id == start_snapshot_id:
```

Comment:
It seems snapshot_id can never be start_snapshot_id, because of the condition `current_snapshot.snapshot_id > start_snapshot_id`?

Reply:
Refactor a bit, actually all_snapshot_ids will start with start_snapshot_id for ancestor check.